src/page-modules/assistant/line-filter/index.tsx

```diff
@@ -2,44 +2,69 @@ import LabeledInput, { LabeledInputProps } from '@atb/components/labled-input';
 import { Typo } from '@atb/components/typography';
 import { getOrgData } from '@atb/modules/org-data';
 import { PageText, useTranslation } from '@atb/translations';
-import { ChangeEvent, useState } from 'react';
+import { ChangeEvent, useEffect, useState } from 'react';
 import style from './line-filter.module.css';
+import useSWR from 'swr';
 type LineFilterProps = {
   filterState: string[] | null;
   onChange: (lineFilter: string[] | null) => void;
 };
 
+const fetcher = (url: string) => fetch(url).then((res) => res.json());
+
 export default function LineFilter({ filterState, onChange }: LineFilterProps) {
   const { t } = useTranslation();
 
-  const { orgLineIdPrefix } = getOrgData();
-  const [error, setError] = useState<LabeledInputProps['validationError']>();
-  const [localFilterState, setLocalFilterState] = useState(
-    filterState?.map((line) => line.replace(orgLineIdPrefix, '')).join(', ') ??
-      '',
+  const { authorityId } = getOrgData();
+
+  const { data, error, isLoading } = useSWR(
+    `api/assistant/lines/${authorityId}`,
+    fetcher,
   );
+  const [validationError, setValidationError] =
+    useState<LabeledInputProps['validationError']>();
+  const [localFilterState, setLocalFilterState] = useState('');
+  const [publicCodeLineMap, setPublicCodeLineMap] =
+    useState<Map<string, string[]>>();
 
   const onChangeWrapper = (event: ChangeEvent<HTMLInputElement>) => {
+    if (error || isLoading) return;
     const lineFilter = event.target.value;
     setLocalFilterState(lineFilter);
-    setError(undefined);
+    setValidationError(undefined);
 
     if (!isValidFilter(lineFilter)) {
-      setError(t(PageText.Assistant.search.lineFilter.error));
+      setValidationError(t(PageText.Assistant.search.lineFilter.error));
       return;
     }
 
     if (!lineFilter) {
       onChange(null);
     } else {
-      const linesWithPrefix = lineFilter
+      const lines = lineFilter
         .split(',')
-        .map((line) => `${orgLineIdPrefix}${line.trim()}`);
+        .map((line) => publicCodeLineMap?.get(line.trim()))
+        .filter((line): line is string[] => line !== undefined)
+        .flat();
 
-      onChange(linesWithPrefix);
+      onChange(lines);
     }
   };
 
+  useEffect(() => {
+    if (!data) return;
+    setPublicCodeLineMap(new Map(JSON.parse(data.publicCodeLineMapString)));
+  }, [data]);
+
+  useEffect(() => {
+    if (!publicCodeLineMap || localFilterState !== '') return;
+    const publicCodeString = Array.from(publicCodeLineMap.entries())
+      .filter(([_, line]) => filterState?.some((code) => line.includes(code)))
+      .map(([code]) => code)
+      .join(', ');
+    setLocalFilterState(publicCodeString);
+  }, [publicCodeLineMap, localFilterState, filterState]);
```

Comment:
I'm not sure this is the most robust way of solving this. 🤔 Think we need to do some testing, as I was experiencing that when removing filters they popped back in (impossible to remove)